include/inviwo/core/datastructures/volume/volumedisk.h

```diff
@@ -57,8 +59,17 @@ class IVW_CORE_API VolumeDisk : public VolumeRepresentation,
     virtual void setDimensions(size3_t dimensions) override;
     virtual const size3_t& getDimensions() const override;
 
+    /**
+     * \brief update the swizzle mask of the color channels when sampling the volume
+     *
+     * @param mask new swizzle mask
+     */
+    virtual void setSwizzleMask(const SwizzleMask& mask) override;
```

Comment:
Can these be placed in VolumeRepresentation so that not each representation needs to implement the same functionality?

Reply:
The functionality is not the same, and we generally prefer the base class to be pure abstract if possible.